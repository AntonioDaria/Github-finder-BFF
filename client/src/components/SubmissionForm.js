import React, { useState } from 'react'
import Results from "./Results";
import { findFavLanguage } from "../services/search.service";

const SubmissionForm = () => {

    const [text, setText] = useState('');
    const [result, setResults] = useState([]);

    const onSubmit = async (e) => {
        e.preventDefault();

        const userName = text;
        
        const apiResult = await findFavLanguage(userName);

        setResults(apiResult);
    }

    return (
        <>
            <h3>Find out a Github user's favourite language</h3>
            <form onSubmit={onSubmit}>
                <div className="form-control">
                    <label htmlFor="text">Username</label>
                    <input type="text" value={text} onChange={(e) => setText(e.target.value)} placeholder="Enter name..." />
                </div>
                <button className="btn">Search User</button>
            </form>
            <br></br>
            <Results results={result} />
        </>
    )
}

export default SubmissionForm