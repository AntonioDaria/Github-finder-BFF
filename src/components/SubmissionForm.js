import React, { useState } from 'react'

const SubmissionForm = () => {

    //const [result, setResults] = useState([]);
    const [text, setText] = useState('');

    const onSubmit = async (e) => {
        e.preventDefault();


    }

    return (
        <>
            <h3>Find out a Github user favourite language</h3>
            <form onSubmit={onSubmit}>
                <div className="form-control">
                    <label htmlFor="text">Username</label>
                    <input type="text" value={text} onChange={(e) => setText(e.target.value)} placeholder="Enter name..." />
                </div>
                <button className="btn">Search User</button>
            </form>
            <br></br>
            {/* <Results results={result} /> */}
        </>
    )
}

export default SubmissionForm