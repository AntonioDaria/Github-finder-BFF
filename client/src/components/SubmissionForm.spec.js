import React from 'react';
import { fireEvent, render } from '@testing-library/react';
import SubmissionForm from './SubmissionForm.js'
import { act } from "react-dom/test-utils";
import { findFavLanguage } from '../services/search.service'

jest.mock('../services/search.service');

describe('Submission Form component', () => {
  let container;

  beforeEach(() => {
    ({ container } = renderSubmissionForm());
  });

  it('should match the snapshot', () => {
    expect(container).toMatchSnapshot();
  });

  describe('OnSubmit', () => {
    it('should call findFavLanguage when clicking the submit button.', async () => {
      let getAllByText;
      ({ getAllByText } = renderSubmissionForm());

      const button = getAllByText("Search User");

      act(() => {
        fireEvent.click(button[0]);
      });

      expect(findFavLanguage).toHaveBeenCalledTimes(1);

    });
  });
});

const renderSubmissionForm = () => {
  return render(
    <SubmissionForm />
  );
};