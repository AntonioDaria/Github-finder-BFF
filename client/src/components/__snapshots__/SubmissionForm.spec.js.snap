// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Submission Form component should match the snapshot 1`] = `
<div>
  <h3>
    Find out a Github user's favourite language
  </h3>
  <form>
    <div
      class="form-control"
    >
      <label
        for="text"
      >
        Username
      </label>
      <input
        placeholder="Enter name..."
        type="text"
        value=""
      />
    </div>
    <button
      class="btn"
    >
      Search User
    </button>
  </form>
  <br />
  <div>
    <h3>
      Our Best Guess is:
    </h3>
    <div
      class="results-container"
    />
  </div>
</div>
`;
